metpy/calc/tools.py

```
@@ -831,3 +832,296 @@ def _less_or_close(a, value, **kwargs):
+
+
+@exporter.export
+def laplacian(f, **kwargs):
+    """Calculate the laplacian of a grid of values.
+
+    Works for both regularly-spaced data, and grids with varying spacing.
+
+    Either `x` or `deltas` must be specified.
+
+    Parameters
+    ----------
+    f : array-like
+        Array of values of which to calculate the derivative
+    x : array-like, optional
+        The coordinate values corresponding to the grid points in `f`
+    deltas : array-like, optional
+        Spacing between the grid points in `f`. There should be one item less than the size
+        of `f` along `axis`.
+
+    Returns
+    -------
+    array-like
+        The laplacian
+
+    See Also
+    --------
+    gradient
+
+    """
+    pos_kwarg, positions = _process_gradient_args(kwargs)
+    return sum(second_derivative(f, axis=ind, **{pos_kwarg: positions[ind]})
+               for ind, pos in enumerate(positions))
+
+
+def _broadcast_to_axis(arr, axis, ndim):
+    """Handle reshaping coordinate array to have proper dimensionality.
+
+    This puts the values along the specified axis.
+    """
+    if arr.ndim == 1 and arr.ndim < ndim:
+        new_shape = [1] * ndim
+        new_shape[axis] = arr.size
+        arr = arr.reshape(*new_shape)
+    return arr
+
+
+def _process_gradient_args(kwargs):
+    """Handle common processing of arguments for gradient and gradient-like functions."""
+    if 'deltas' in kwargs:
+        return 'delta', kwargs['deltas']
+    elif 'x' in kwargs:
+        return 'x', kwargs['x']
+    else:
+        raise ValueError('Must specify either "x" or "delta" for value positions.')
+
+
+def _process_deriv_args(f, kwargs):
+    """Handle common processing of arguments for derivative functions."""
+    n = f.ndim
+    axis = normalize_axis_index(kwargs.get('axis', 0), n)
+
+    if f.shape[axis] < 3:
+        raise ValueError('f must have at least 3 point along the desired axis.')
+
+    if 'delta' in kwargs:
```

Similar question here on silently ignoring an argument if both are passed.

---

ok